src/views/ViewModule/view-module.js

```diff
@@ -0,0 +1,125 @@
+/* eslint-disable consistent-return */
+/* eslint-disable react/destructuring-assignment */
+import React, { Component } from 'react';
+import { Container, Row, Col } from 'reactstrap';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import { Link, Redirect } from 'react-router-dom';
+import ModuleSidebar from '../../components/ModuleSidebar';
+import Spinner from '../../components/Spinner/Spinner';
+import Navbar from '../../components/Header/Header';
+import Button from '../../components/Button';
+import Footer from '../../components/Footer/Footer';
+import { getModule } from '../../../store/actions/moduleActions';
+import './view-module.scss';
+
+export class ViewModule extends Component {
+  componentDidMount() {
+    const { moduleId } = this.props.match.params;
+    this.props.getModule(moduleId);
+  }
+
+  render() {
+    let allContents;
+
+    if (!this.props.isAuthenticated) {
+      return <Redirect to="/login" />;
+    }
+
+    const { id, name, description, contents } = this.props.singleModule;
+
+    if (contents) {
+      allContents = contents.map((content) => {
```

Comment:
Why can't we have this as a separate component on its own that takes in a list of content as props? 🤔

Reply:
Sure, that can be done.